ui/lib/apps/Statement/utils/useStatementTableController.ts

```diff
@@ -141,9 +141,11 @@ export default function useStatementTableController(
     const { schemas, stmtTypes, searchText, timeRange } = queryOptions
     const cacheKey = `${schemas.join(',')}_${stmtTypes.join(
       ','
-    )}_${searchText}_${stringifyTimeRange(timeRange)}_${selectedFields}`
+    )}_${searchText}_${stringifyTimeRange(timeRange)}_${
```

Comment:
This just makes the cache ineffective: when returning back, the data may be changed. I think what you should do is to set the time picker to the actual memorized time when returning back.

What do you think @baurine

Reply:
The logic of the time range calculation takes the nearest whole point or half point. For example, for recent 30 mins, 8:12 would be calculated as 8:00 - 8:30 time range, so the cache is still effective.

And the behavior is more in accordance with the "recent" - When we use the time range of the recent x minutes, the data should be the recent x minutes.

Reply:
I think how time range calculates doesn't matter. From a user perspective, user is supposed to always see a UI with corresponding data. With cache, we provides a feature that after user visits a record and goes back to the list view the user should see the list data exactly as the previous one. This two should be the primary principles that guides all behaviours.

Reply:
Under these two principles, after visiting a detail page and goes back, user should still seeing the previous list, in all cases, no matter when and how. And what we need to ensure is that all UI controls are correctly reflecting this list. For example, if this list is the 10:00 ~ 10:30 time range then the UI should indicate this range (if the last selected "recent XXX" no longer matches).

Reply:
As an alternative, it would also be great if we give user hints (while keeping the UI controls), like "This list is outdated. To see latest data, click "Refresh").

Reply:
I got you. Refreshing data needs to be done manually.

And each response of the data from the server needs to be cached with a time range for detail plans queries. I need to recover this part of the code...

Reply:
This would be great.

Reply:
Agree with @breeswish , this change would make the cache ineffective, and it's better to show the memoried time range in the time picker when backing from the detail page.

And for the caching strategy, currently we implement it manually, not sure whether react-query or swr would help us simplify the logic.